select:add command-line flag -search.maxBinaryOpPushdownLabelValues

tydhot opened the pull request:
### Describe Your Changes

Binary operations like `exprFirst op exprSecond` in VictoriaMetrics are performed in the following way:
1. Execute exprFirst.
2. Extract **common label filters** from the result of step 1.
3. Apply these common label filters to `exprSecond` and execute it, in order to retrieve less time series from vmstorage nodes.

In step 2, only labels with less than `100` (hard-coded) value could be used as **common label filter** (e.g. `{common_lb=~"v1|v2|...|v100"}`.

In our scenarios, a label, take `instance` label as an example, could has thousands of candidate values. Regarding bring more pressure to vmstorage node, it's still beneficial if labels with more than 100 values could be used as filter in `exprSecond`, with enough vmstorage resources. After adjusting the value from `100` to `10000`, our query round-trip time drops significantly from 5s to 2s. 

This pull request change the hard-coded value into a configurable flag.

### Checklist

The following checks are **mandatory**:

- [x] My change adheres [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/contributing/).

jiekun commented:
Thanks for the contribution and patience. I'll be back checking this shortly.

jiekun commented:
@tydhot, please review my commits at https://github.com/VictoriaMetrics/VictoriaMetrics/pull/7243/commits as the following changes have been made:
1. Renamed the flag to `-search.maxBinaryOpPushdownLabelValues`. This can still be changed based on feedback from other reviewers.
2. Added CHANGELOG and flag documentation in `docs/Cluster-VictoriaMetrics.md`. This step is mandatory when introducing a new feature and flag.
3. The description of this pull request has been updated to provide more context on this push-down common filter.

tydhot commented:
> @tydhot, please view my commits at https://github.com/VictoriaMetrics/VictoriaMetrics/pull/7243/commits as the following changes have been made:
> 
> 1. Renamed the flag to `-search.maxBinaryOpPushdownLabelValues`. This can still be changed based on feedback from other reviewers.
> 2. Added CHANGELOG and flag documentation in `docs/Cluster-VictoriaMetrics.md`. This step is mandatory when introducing a new feature and flag.
> 3. The description of this pull request has been updated to provide more context on this push-down common filter.

thanks！ lgtm

f41gh7 reviewed (approved):
LGTM

f41gh7 commented:
Thanks for contribution!
